### PR TITLE
Rename fields in taker request

### PR DESCRIPTION
**Pull request by feuGeneA**
In lieu of CI, a log of a test run on the latest commit:
```
14May17:56:34 [2 jobs] ~/dev/quote-server[feat/rename-indicative-quote-params=]$ git log -1
commit 4d16e888ae43b292dc02f91f670006217627ed41
Author: F. Eugene Aumson <feuGeneA@users.noreply.github.com>
Date:   Thu May 14 17:56:24 2020 -0400

    Add .prettierignore file
14May17:56:37 [2 jobs] ~/dev/quote-server[feat/rename-indicative-quote-params=]$ git status
On branch feat/rename-indicative-quote-params
Your branch is up-to-date with 'origin/feat/rename-indicative-quote-params'.
nothing to commit, working tree clean
14May17:56:39 [2 jobs] ~/dev/quote-server[feat/rename-indicative-quote-params=]$ yarn clean && yarn build && yarn test && yarn lint
yarn run v1.19.1
$ shx rm -rf lib
Done in 0.13s.
yarn run v1.19.1
$ tsc -b
Done in 2.33s.
yarn run v1.19.1
$ NODE_ENV=test mocha --require source-map-support/register --require make-promises-safe ./lib/test/**/**/*_test.js --exit


  api key handler
    ✓ reject when canMakerControlSettlement == false & no API key specified
    ✓ accept when canMakerControlSettlement == false & API Key specified
    ✓ accept when canMakerControlSettlement == true & no API Key specified
    ✓ accept when canMakerControlSettlement == true & API Key specified

  taker request handler
    ✓ should defer to quoter and return response for firm quote
    ✓ should succeed without an API key if `canMakerControlSettlement` is set to `true` when requesting a firm quote
    ✓ should defer to quoter and return response for indicative quote
    ✓ should handle empty indicative quote
    ✓ should handle empty firm quote
    ✓ should invalidate a bad request

  submit request handler
    ✓ should defer to quoter and return response for submit request
    ✓ should succeed without an API key when submitting a fill
    ✓ should invalidate a bad request
    ✓ should handle empty indicative quote


  14 passing (121ms)

Done in 0.62s.
yarn run v1.19.1
$ run-p tslint prettier:lint
$ tslint --format stylish --project .
$ yarn prettier --list-different
$ prettier '**/*.{ts,tsx,json,md}' --config .prettierrc --list-different
no-unused-variable is deprecated. Since TypeScript 2.9. Please use the built-in compiler checks instead.
Done in 3.50s.
```

### CODE REVIEW

File: CHANGELOG.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[
+    {
+        "version": "0.1.0",
```
Comment (steveklebanoff) [inline CHANGELOG.json:3]: sup w/ these duplicate entries?

File: package.json [hunk 2]
```diff
@@ -10,12 +10,15 @@
         "build": "tsc -b",
         "watch": "tsc --watch",
         "clean": "shx rm -rf lib",
-        "lint": "tslint --format stylish --project .",
-        "lint:fix": "tslint --format stylish --project . --fix",
-        "prettier": "prettier --write '**/*.{ts,tsx,json,md}' --config .prettierrc",
-        "prettier:ci": "prettier --list-different '**/*.{ts,tsx,json,md}' --config .prettierrc",
+        "tslint": "tslint --format stylish --project .",
+        "tslint:fix": "yarn tslint --fix",
+        "prettier": "prettier '**/*.{ts,tsx,json,md}' --config .prettierrc",
```
Comment (steveklebanoff) [inline package.json:15]: I know we've had this convo before, but I personally would like to keep our `script` patterns the same as the monorepo.

If you feel strongly about grouping prettier & tslint into one "lint" command I see your point, but I do think it is a bit strange to have different patterns between this repo & the monorepo

Comment (feuGeneA) [reply package.json:15]: I appreciate your effort to keep us consistent.  FYI I'm a crusade to do this everywhere now.  Every time work in a monorepo package I'm updating it to include prettier in lint.  Though, so far I've only done it to asset-swapper and migrations :)

File: CHANGELOG.json (hunk 1 above)
Comment (feuGeneA) [reply CHANGELOG.json:12]: Thanks, addressed in https://github.com/0xProject/quote-server/pull/5/commits/598b8f91a808030f5ff59177ad11a1a328169b55

File: package.json (hunk 2 above)
Comment (steveklebanoff) [reply package.json:15]: 🙀

File: CHANGELOG.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[
```
Comment (steveklebanoff) [inline CHANGELOG.json:1]: Does this CHANGELOG.json actually do anything?  I was expecting a `publish` entry in `scripts` to investigate, but I don't see one 🤔

Comment (feuGeneA) [reply CHANGELOG.json:1]: Nope, it's just me keeping a manual record.  Having it actually do something would be great, but I figured doing things manually would be better than doing nothing at all.

File: src/types.ts [hunk 4]
```diff
@@ -8,27 +8,38 @@ type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T
...
+export interface RFQTFirmQuote {
     signedOrder: SignedOrder;
-    quoteExpiry: number; // If RFQT order, simply set to order expiry
 }
 
+export interface RFQMFirmQuote extends RFQTFirmQuote {
+    quoteExpiry: number;
+}
+
+export type FirmQuote = RFQTFirmQuote | RFQMFirmQuote;
```
Comment (steveklebanoff) [inline src/types.ts:41]: Lookin' good 👍

**Pull request merged**